### PR TITLE
Split integration tests into jobs

### PR DESCRIPTION
**Pull request by teresaromero**
# Proposed changes

Fixes https://okteto.atlassian.net/browse/DEV-507

- remove integration commands, instead run jobs for each of them, this way we can retrigger a job without having to retrigger the whole workflow
- when labels on PRs are used to run integration, the new workflows are triggered without duplicating the `lint-test-build` launched by the push to the branch
- use okteto context instead of deprecated okteto login on integration jobs and okteto analytics --disable instead of creating the file
- reduce binary build time by using exclusive cache key, when building the windows binary the pkg where not saved at the cache key used on that job (reduced 5 minuts)

How to test:
- create new branch from this and make some code changes, push and check workflows
- trigger with `run-e2e` labels (unix and windows) the integration workflows are triggered
- create a branch that ends with win or e2e to check workflows are triggered for integration

### CODE REVIEW

File: .circleci/config.yml
Comment (teresaromero) [reply .circleci/config.yml:136]: indeed, the changed from commands to jobs, so this setup is needed on each job. i will try one more time to see if we can optimize this.

Not sure what you mean with shared envvars at the executor, but we need to setup these at the test as is part of the test steps to "select the binary" which is built from the current commit in the pipeline

Comment (ifbyol) [reply .circleci/config.yml:141]: Within each executor defined under `executors`, you can define a set of environment variables to be used when that executor is configured: https://circleci.com/docs/configuration-reference/#executors. Fran means to move the environment variable defined on each job to the executor used on those jobs, so they are only defined once

Comment (teresaromero) [reply .circleci/config.yml]: i've re-done by using workflows and sharing the .okteto folder to use settings.

@maroshii @ifbyol you can take a look now.

**Review by ifbyol — COMMENTED** [review]
As we discussed offline, we might try to have more specific workflows instead of having "generic" workflows with conditions in the workflow and the jobs for simplicity. Let's try that to see if that make things simpler and easier to follow. As it is know, it might be hard to know when a specific job is triggered

File: .circleci/config.yml
Comment (teresaromero) [reply .circleci/config.yml]: nop, i missed those, thanks

File: .circleci/config.yml
Comment (teresaromero) [reply .circleci/config.yml]: yes, done

File: .circleci/config.yml
Comment (teresaromero) [reply .circleci/config.yml:662]: adding `when` to the workflow intended to replace the filters on each job, so this pattern was intended to be the "match" to exclude any tag, but now i see it might not be correct. 

```
filters:
            branches:
              only: *release-branch-regex
            tags:
              ignore: /.*/ <-- this one, to ignore any tag? 
```

File: .circleci/config.yml
Comment (ifbyol) [inline .circleci/config.yml:654]: hmm, this is when we create or update release branches? In those case, we don't use tags. I think the comment is not accurate, right? I mean, I think the condition in the `when` is right,  but this comment says the opposite, right?

**Review by ifbyol — APPROVED** [review]
It looks good but I'd wait for at least another approval as this is something common and all of us should be confortable with it

File: .circleci/config.yml
Comment (jLopezbarb) [inline .circleci/config.yml:375]: Can we rename them to `e2e` instead of `integration`? We started changing the name from integration to e2e but have to stop in the middle of the refactor

File: .circleci/config.yml
Comment (jLopezbarb) [inline .circleci/config.yml:555]: Probably no in this PR but, should we split the workflows so we don't have so many `filters.branch.only=master` 
At least we should differentiate between the one for the commit and the one for the merge to master

**Review by jLopezbarb — APPROVED** [review]

File: .circleci/config.yml
Comment (teresaromero) [reply .circleci/config.yml:555]: I've filed an issue in order to work on this sometime https://okteto.atlassian.net/browse/DEV-534

**Pull request merged**
